### PR TITLE
chore(error handling): remove bincode's error type as a public type

### PR DESCRIPTION
**Pull request by Dylan-DPC**
converts bincode error type to one of the custom types for each case

standardises some of the returned result types and creates/edits a few error subtypes

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -6,6 +6,8 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
+use crate::agreement::ProposalError;
+use crate::messages::{CreateError, ExtendProofChainError};
```
Comment (maqi) [inline src/error.rs:10]: better use
```rust
use crate::{
    agreement::ProposalError,
    messages::{CreateError, ExtendProofChainError},
    section::SectionChainError,
};
```
to group

File: src/messages/mod.rs [hunk 2]
```diff
@@ -267,7 +273,8 @@ impl Message {
             dst: &self.dst,
             dst_key: self.dst_key.as_ref(),
             variant: &self.variant,
-        })?;
+        })
+        .map_err(|_| Error::MessageNotSigned)?;
```
Comment (maqi) [inline src/messages/mod.rs:277]: this seems not reflecting the correct reason of failure?

File: src/error.rs [hunk 3]
```diff
@@ -42,6 +44,16 @@ pub enum Error {
     InvalidSectionChain(#[from] SectionChainError),
     #[error("Messaging protocol error: {0}")]
     Messaging(#[from] sn_messaging::Error),
+    #[error("Message not signed")]
+    MessageNotSigned,
```
Comment (maqi) [inline src/error.rs:48]: Is this being used anywhere?

Comment (Dylan-DPC) [reply src/error.rs]: removed it

**Review by maqi — DISMISSED** [review]

**Review by maqi — APPROVED** [review]

**Review by Yoga07 — APPROVED** [review]
LGTM! 👍

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**